libs/onboarding-sdk-product/src/main/java/it/pagopa/selfcare/product/service/ProductServiceDefault.java

```diff
@@ -154,6 +155,27 @@ public Product getProductIsValid(String productId) {
         return getProduct(productId, true);
     }
 
+    /**
+     * The validateProductRole function is used to validate a product role searching for it within the role map for a specific product
+     * given productId. It returns founded ProductRole object filtered by productRole code
+     */
+    @Override
+    public ProductRole validateProductRole(String productId, String productRole, PartyRole role) {
+        if (role == null) {
+            throw new IllegalArgumentException("Role is mandatory to check productRole");
+        }
+
+        Product product = getProduct(productId);
+        ProductRoleInfo productRoleInfo = product.getRoleMappings().get(role);
```

Comment:
you could check if getRoleMappings() is not null or something like this **Optional.ofEmpty(product.getRoleMappings()).map(...).orElse(null)**